client/verta/tests/custom_modules/test_custom_modules.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+
+import filecmp
+import json
+import os
+import pkgutil
+import zipfile
+
+import hypothesis
+import pytest
+import six
+
+from verta.tracking.entities._deployable_entity import _DeployableEntity
+from verta._internal_utils.custom_modules import CustomModules
+
+from .. import strategies, utils
+from . import contexts
+
+
+class TestCollectPipInstalledModule:
+    @staticmethod
+    def assert_in_custom_modules(custom_modules, module_name):
+        module = CustomModules.get_module_path(module_name)
+
+        with utils.tempdir() as custom_modules_dir:
+            with zipfile.ZipFile(custom_modules, "r") as zipf:
+                zipf.extractall(custom_modules_dir)
+
+            # TODO: extract sys.path from _verta_config.py instead of walking
+            for parent_dir, dirnames, filenames in os.walk(custom_modules_dir):
+                if os.path.basename(module) in dirnames + filenames:
+                    retrieved_module = os.path.join(
+                        parent_dir,
+                        os.path.basename(module),
+                    )
+                    break
+            else:
+                raise ValueError("module not found in custom modules")
+
+            if os.path.isfile(module):
+                assert filecmp.cmp(module, retrieved_module)
+            else:
+                utils.assert_dirs_match(module, retrieved_module)
+
+    @pytest.mark.parametrize(
+        "name",
+        sorted(module[1] for module in pkgutil.iter_modules()),
+    )
+    def test_module(self, name):
```

Comment:
This test has a number of cases that I'm skipping (see the `if`s just below); the failures are unrelated to this fix, and are outside this ticket's scope to resolve 😞